### ISSUE TITLE
sparse_softmax_cross_entropy_with_logits gives NaN instead of error when using non-existent labels

### ISSUE THREAD

**Issue by mtanti** [tensorflow/tensorflow#10268]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 14.04

- **TensorFlow installed from (source or binary)**:
Source

- **TensorFlow version (use command below)**:
v1.0.0-65-g4763edf-dirty
1.0.1

- **Bazel version (if compiling from source)**:
0.4.5

- **CUDA/cuDNN version**:
V8.0.61

- **GPU model and memory**:
NVIDIA GFORCE GTX 760 2GB

- **Exact command to reproduce**:
import tensorflow as tf
sess = tf.Session(0
sess.run(tf.nn.sparse_softmax_cross_entropy_with_logits(labels=[ 100 ], logits=[[ 0.0, 1.0 ]]))
>>> array([ nan], dtype=float32)

### Describe the problem
Running above code gives NaN instead of an error when on Ubuntu but when I run the same code on Windows I correctly get an InvalidArgumentError error.

### Source code / logs
See code above.

**Comment by jart** [comment]
This behavior is documented and working as intended:

```
    labels: `Tensor` of shape `[d_0, d_1, ..., d_{r-1}]` (where `r` is rank of
      `labels` and result) and dtype `int32` or `int64`. Each entry in `labels`
      must be an index in `[0, num_classes)`. Other values will raise an
      exception when this op is run on CPU, and return `NaN` for corresponding
      loss and gradient rows on GPU.
```

It's probably not a behavior we could change, due to our [API stability promise](https://www.tensorflow.org/programmers_guide/version_semantics).
